kitsune/sumo/static/sumo/scss/form-wizard.styles.scss

```diff
@@ -94,6 +94,15 @@ progress::-webkit-progress-value {
   background-color: var(--color-progress);
 }
 
+progress::-moz-progress-bar {
+  transition: padding-bottom 0.6s;
```

Comment:
Generally speaking, I think we try to avoid this kind of transition because it's not something that runs on the compositor (aka, it's not GPU accelerated). That means if, during the animation, some event on the main thread causes us to blow out the 16ms frame budget, then the animation will stutter.

The CSS transitions that we _can_ use on the GPU are the matrix-y ones like `translate` and `scale`. Those ones we can run smoothly in almost all scenarios, even if the main thread janks a little.

This is, I believe, how the about:welcome progress bar works. If you open about:welcome and examine it with the Inspector, there's a container element and the indicator itself, and the indicator is translated horizontally based on a `--progress-bar-progress` CSS variable, using this calculation:

```
transition: var(--progress-bar-transition);
translate: calc(var(--progress-bar-progress, 0%) - 100%);
```

where `--progress-bar-transition` is `none` in the prefers-reduced-motion case, and `--progress-bar-transition: 0.6s translate;` otherwise.

I suspect we might want to go this route, mainly because I think Tati expected the progress bar to behave the same as about:welcome's. FWIW, this sort of thing might make a good reusable component.

Reply:
Thanks! That's super interesting and good to know. I was kind of just fixated on avoiding the two `<div>` approach in favor of using the built in `<progress>` element, but that's maybe not worth the hassle. The `about:welcome` progress bar also has some nice `aria-` attrs we should probably add here too. May keep tinkering for a bit.

This would be a great candidate for a reusable component - there's at least one other progress bar I can think of off the top of my head in Firefox view.